client/my-sites/earn/memberships/index.jsx

```diff
@@ -518,6 +507,49 @@ class MembershipsSection extends Component {
 		);
 	}
 
+	renderNotices() {
+		const { siteSlug, translate } = this.props;
+		const stripe_connect_success = this.props?.query?.stripe_connect_success;
+
+		if ( stripe_connect_success === 'earn' ) {
```

Comment:
'earn' => Shouldn't that be calypso?

I haven't tested, but that seems like it should break the original flow by not diaplaying the notice....

Can we introduce constants for that as well?

Reply:
> I haven't tested, but that seems like it should break the original flow by not diaplaying the notice....

It shouldn't. Once we pass `calypso` as a query param for `source`,  `earn` will be returned as a query param for the redirect. See D103932-code for more context.

Reply:
> Can we introduce constants for that as well?

We could! I didn't add it as a const because it's only being used in one file.

Reply:
> We could! I didn't add it as a const because it's only being used in one file.

Oh, ok...

Reply:
> It shouldn't. Once we pass calypso as a query param for source, earn will be returned as a query param for the redirect. See D103932-code for more context.

Ok, it kind of makes sense now. But this is not very straightforward....